### ISSUE TITLE
spark-sql fails to run query with exception 'can't serialize class org.apache.spark.sql.types.UTF8String'

### ISSUE THREAD

**Issue by wonnyz** [Stratio/spark-mongodb#6]
It fails when string predicate is used in WHERE clause.

```
$ bin/spark-sql --jars lib/spark-mongodb-core-0.8.4.jar,lib/casbah-core_2.10-2.8.0.jar,lib/casbah-commons_2.10-2.8.0.jar,lib/casbah-query_2.10-2.8.0.jar,lib/mongo-java-driver-2.13.0.jar

SET spark.sql.hive.version=0.13.1
spark-sql> create temporary table test using com.stratio.provider.mongodb options (host 'localhost', database 'test_log', collection 'logs');
Time taken: 4.119 seconds
spark-sql> select distinct user, type from test;
GOOGLE_PLAY_STORE       1600
GOOGLE_PLAY_STORE       12000
...
Time taken: 1.589 seconds, Fetched 59 row(s)
spark-sql> select aid, user, type from test where user = 'GOOGLE_PLAY_STORE' limit 2;
15/06/24 20:15:31 ERROR Executor: Exception in task 0.0 in stage 6.0 (TID 205)
java.lang.IllegalArgumentException: can't serialize class org.apache.spark.sql.types.UTF8String
        at org.bson.BasicBSONEncoder._putObjectField(BasicBSONEncoder.java:299)
        at org.bson.BasicBSONEncoder.putObject(BasicBSONEncoder.java:194)
        at org.bson.BasicBSONEncoder.putObject(BasicBSONEncoder.java:136)
        at com.mongodb.DefaultDBEncoder.writeObject(DefaultDBEncoder.java:36)
        at com.mongodb.OutMessage.putObject(OutMessage.java:289)
        at com.mongodb.OutMessage.writeQuery(OutMessage.java:211)
        at com.mongodb.OutMessage.query(OutMessage.java:86)
        at com.mongodb.DBCollectionImpl.find(DBCollectionImpl.java:81)
        at com.mongodb.DBCollectionImpl.find(DBCollectionImpl.java:66)
        at com.mongodb.DBCursor._check(DBCursor.java:498)
        at com.mongodb.DBCursor._hasNext(DBCursor.java:621)
        at com.mongodb.DBCursor.hasNext(DBCursor.java:657)
        at com.mongodb.casbah.MongoCursorBase$class.hasNext(MongoCursor.scala:76)
        at com.mongodb.casbah.MongoCursor.hasNext(MongoCursor.scala:517)
        at com.stratio.provider.mongodb.reader.MongodbReader$$anonfun$hasNext$2.apply(MongodbReader.scala:58)
        at com.stratio.provider.mongodb.reader.MongodbReader$$anonfun$hasNext$2.apply(MongodbReader.scala:58)
        at scala.Option.fold(Option.scala:157)
        at com.stratio.provider.mongodb.reader.MongodbReader.hasNext(MongodbReader.scala:58)
        at com.stratio.provider.mongodb.rdd.MongodbRDDIterator.hasNext(MongodbRDDIterator.scala:57)
        at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:327)
        at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:327)
        at scala.collection.Iterator$$anon$14.hasNext(Iterator.scala:388)
        at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:327)
        at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:308)
        at scala.collection.Iterator$class.foreach(Iterator.scala:727)
        at scala.collection.AbstractIterator.foreach(Iterator.scala:1157)
        at scala.collection.generic.Growable$class.$plus$plus$eq(Growable.scala:48)
        at scala.collection.mutable.ArrayBuffer.$plus$plus$eq(ArrayBuffer.scala:103)
...
```

the query finished successfully when I used number in WHERE clause.

```
spark-sql> select aid, user, type from test where type = 1600 limit 2;
54dca26084a92c48da000679        GOOGLE_PLAY_STORE       1600
546c2cc17275bca41c000017        NULL    1600
Time taken: 0.287 seconds, Fetched 2 row(s)
spark-sql>
```

I'm testing with connector version 0.8.4, spark 1.4.0 (prebuilt for hadoop 2.4 and above)

**Comment by derfred** [comment]
a similar error was reported here:
https://github.com/spirom/spark-mongodb-connector/issues/18

**Comment by pmadrigal** [comment]
We have updated master branch fixing this error.

Thank you for your feedback!

**Comment by pmadrigal** [comment]
Now it is available in Maven Central as well (version 0.8.5).

Could you close the issue if it is ok?

Thank you for your feedback!

**Comment by wonnyz** [comment]
fixed. thank you!
